user_tools/src/spark_rapids_pytools/cloud_api/azurestorage.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Implementation of Azure Data Lake Storage with ABFS (Azure Blob File System) related functionalities."""
+
+from dataclasses import dataclass
+
+from spark_rapids_pytools.cloud_api.sp_types import CMDDriverBase
+from spark_rapids_pytools.common.sys_storage import StorageDriver, FSUtil
+
+
+@dataclass
+class AzureStorageDriver(StorageDriver):
+    """
+    Wrapper around azure commands such as copying/moving/listing files.
+    """
+    cli: CMDDriverBase
+
+    @classmethod
+    def get_cmd_prefix(cls):
+        pref_arr = ['azcopy']
```

Comment:
Can we have a comment on the class level explaining why we are using `azcopy`, and what are the assumptions on how the authentication works (i.e., expects that the credential file to have a token..bla..bla)?

What is the difference between azcopy and other commands such "az storage" [data-lake-storage-directory-file-acl-cli](https://learn.microsoft.com/en-us/azure/storage/blobs/data-lake-storage-directory-file-acl-cli)

Reply:
Yes, I will add a comment on this point.
To use `az storage fs file`, we need to have permission to be assigned to role "Storage Blob Data Contributor". For `azcopy`, we can work around this permission with a [SAS token](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10?toc=%2Fazure%2Fstorage%2Fblobs%2Ftoc.json&bc=%2Fazure%2Fstorage%2Fblobs%2Fbreadcrumb%2Ftoc.json#option-2-use-a-sas-token).